api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -122,27 +122,30 @@
    *
    * @param conflictDetectionFilter an expression on rows in the table
    * @return this for method chaining
+   * @deprecated this will be removed in 0.14.0;
+   *             use {@link #validateNoConflictingOperations(Expression)} instead
    */
-  OverwriteFiles validateNoConflictingAppends(Expression conflictDetectionFilter);
+  @Deprecated
+  default OverwriteFiles validateNoConflictingAppends(Expression conflictDetectionFilter) {
```

Comment:
I am not a fan of renaming but I think the old name would be really confusing now.

Reply:
Agreed.

Reply:
I changed this part as there are cases when we want to validate delete files but not data files. For example, a merge-on-read UPDATE with snapshot isolation. Now we have two methods and no renaming.